src/LinAlg/matrix.h

```diff
@@ -1007,6 +1007,8 @@ namespace utl //! General utility classes and functions.
                                     unsigned int ofsy, int stridey)
   {
     size_t n = this->size() < X.size() ? this->size() : X.size();
+    if (n == 0)
+      return *this;
```

Comment:
better with
```
if (n > 0)
  cblas_...(...);
```
such that you don't need the extra return. Same with the following 3 changes below.

Reply:
ok